lib/apiservers/engine/backends/errors.go

```diff
@@ -47,6 +47,11 @@ func VolumeCreateNotFoundError(msg string) error {
 	return derr.NewErrorWithStatusCode(fmt.Errorf("No volume store named (%s) exists", msg), http.StatusInternalServerError)
 }
 
+// VolumeNotFoundError returns a 404 docker error for a volume get request.
```

Comment:
Does this trigger volume creation in the client in the same way image 404 triggers pull?
IIRC that on-demand creation was implemented explicitly, but don't recall hearing that we checked for implicit behaviours.